client/my-sites/term-selector/terms.jsx

```diff
@@ -0,0 +1,389 @@
+		}
+	},
+
+	hasNoSearchResults() {
+		return ! this.props.loading &&
+			( this.props.terms && ! this.props.terms.length ) &&
+			this.state.searchTerm;
+	},
+
+	hasNoTerms() {
+		return ! this.props.loading && ( this.props.terms && ! this.props.terms.length );
+	},
+
+	getItem( index ) {
+		if ( this.props.termsHierarchy ) {
+			return this.props.termsHierarchy[ index ];
+		}
+	},
+
+	isCompact() {
+		if ( ! this.props.terms || this.state.searchTerm ) {
+			return false;
+		}
+
+		return this.props.terms.length < this.props.searchThreshold;
+	},
+
+	isRowLoaded( { index } ) {
+		return this.props.lastPage || !! this.getItem( index );
+	},
+
+	getItemHeight( item ) {
+		if ( item && this.itemHeights[ item.ID ] ) {
+			return this.itemHeights[ item.ID ];
+		}
+
+		let height = ITEM_HEIGHT;
+
+		if ( item && item.items ) {
+			height += item.items.reduce( ( memo, nestedItem ) => {
+				return memo + this.getItemHeight( nestedItem );
+			}, 0 );
+		}
+
+		return height;
+	},
+
+	getRowHeight( { index } ) {
+		return this.getItemHeight( this.getItem( index ) );
+	},
+
+	getCompactContainerHeight() {
+		return range( 0, this.getRowCount() ).reduce( ( memo, index ) => {
+			return memo + this.getRowHeight( { index } );
+		}, 0 );
+	},
+
+	getResultsWidth() {
+		const { selectorRef } = this.state;
+		if ( selectorRef ) {
+			return selectorRef.clientWidth;
+		}
+
+		return 0;
+	},
+
+	getRowCount() {
+		let count = 0;
+
+		if ( this.props.termsHierarchy ) {
+			count += this.props.termsHierarchy.length;
+		}
+
+		if ( ! this.props.lastPage || this.hasNoSearchResults() ) {
+			count += 1;
+		}
+
+		return count;
+	},
+
+	maybeFetchNextPage() {
+		if ( this.props.loading || ! this.props.terms ) {
+			return;
+		}
+
+		this.props.onNextPage();
+	},
+
+	onSearch( event ) {
+		const searchTerm = event.target.value;
+		if ( this.state.searchTerm && ! searchTerm ) {
+			this.props.onSearch( '' );
+		}
+
+		if ( searchTerm === this.state.searchTerm ) {
+			return;
+		}
+
+		if ( ! this.hasPerformedSearch ) {
+			this.hasPerformedSearch = true;
+			analytics.ga.recordEvent( this.props.analyticsPrefix, 'Performed Term Search' );
+		}
+
+		this.setState( { searchTerm } );
+		this.debouncedSearch();
+	},
+
+	getSelectedIds( selected ) {
+		const selectedObjects = selected || this.props.selected;
+		return selectedObjects.map( function( item ) {
+			if ( ! item.ID ) {
+				return item;
+			}
+
+			return item.ID;
+		} );
+	},
+
+	renderItem( item ) {
+		const onChange = ( ...args ) => this.props.onChange( item, ...args );
+		const setItemRef = ( ...args ) => this.setItemRef( item, ...args );
+
+		const { multiple, defaultTermId, taxonomy, translate } = this.props;
+		const selectedIds = this.getSelectedIds();
+		const itemId = item.ID;
+		const name = unescapeString( item.name ) || translate( 'Untitled' );
+		const checked = includes( selectedIds, itemId );
+		const inputType = multiple ? 'checkbox' : 'radio';
+		const domId = taxonomy + '-option-' + itemId;
```

Comment:
Some of this stuff I couldn't find being used in `<PostSelector />` so I removed. Maybe we can use this as an opportunity to revisit whether some of these attributes are necessary (`id`, `name`)?